mapie/calibration.py

```diff
@@ -0,0 +1,559 @@
+            Calibrator to fit.
+
+        Returns
+        -------
+        Dict[int, RegressorMixin]
+            Dictionnary of fitted calibrators.
+        """
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        max_prob, y_pred = self._get_labels(X)
+        calibrators = {}
+        for item in np.unique(y_pred):
+            calibrator_ = self._fit_calibrator(
+                item,
+                calibrator,
+                y,
+                max_prob,
+                y_pred,
+                sample_weight,
+            )
+            calibrators[item] = calibrator_
+        return calibrators
+
+    def _pred_proba_calib(
+        self,
+        idx: int,
+        item: int,
+        calibrated_values: NDArray,
+        max_prob: NDArray,
+        y_pred: NDArray,
+        calibrators: Dict[int, RegressorMixin],
+    ) -> NDArray:
+        """
+        Using the predicted scores, we calibrate the maximum score with the
+        specifically fitted calibrator. Note that if there is no calibrator
+        for a the specific class, then we simply output the not calibrated
+        values.
+
+        Note that if the calibrated score prediction is 0, there would be an
+        issue when finding the class it belongs to. We set it equal to
+        {np.finfo(np.float64).eps} as this would likely not have large impact
+        on the calibration scores, yet, would set the maximum to the correct
+        label.
+
+
+        Parameters
+        ----------
+        item : int
+            The defined subset to be calibrated.
+        calibrated_values : NDArray
+            Array of calibrated values.
+        max_prob : NDArray of shape (n_samples,)
+            Values to be calibrated.
+        y_pred : NDArray of shape (n_samples,)
+            Indicator of the values to be calibrated.
+        calibrators : Dict[RegressorMixin] of length n_classes
+            Dictionary of all the previously fitted calibrators.
+
+        Returns
+        -------
+        NDArray
+            Updated calibrated values from the predictions of the calibrators.
+
+        Raises
+        ------
+        Warnings
+            If there has not been a fitted calibrator for the specific class.
+        """
+        correct_label = np.where(y_pred.ravel() == item)[0].ravel()
+        if item not in calibrators:
+            calibrated_values[
+                correct_label, idx
+                ] = max_prob[correct_label].ravel()
+            warnings.warn(
+                "WARNING: This predicted label {item} has not been seen during"
+                + " the calibration and therefore scores will remain"
+                + " unchanged."
+            )
+        else:
+            EPSILON = np.finfo(np.float64).eps
+            calibrator_ = calibrators[item]
+            preds_ = calibrator_.predict(max_prob[correct_label])
+            idx_zero_pred = np.where(preds_ < EPSILON)[0]
+            preds_[idx_zero_pred] = EPSILON
```

Comment:
I'm not sure why you believe I should change the code. The aim here is to identify the moments when the predictions are below `EPSILON`. This should identify when the prediction was relatively close to 0 by the calibrator. Hence, if the prediction is too close to 0, it might not be able to identify the correct `np.argmax` in a later context and therefore return the wrong label from the scores.